mantidproject/mantid
# PyChop bugfixes

Fixes some bugs, and makes some modifications in PyChop as identified by instrument scientists:

1. Restricts energy ranges of multi-reps to those with non-negligible neutron fluxes (defined externally as `E>7meV` for Merlin and `E<30meV` for LET.

2. Fixes typos and other bugs in calculations for LET (restricts frame overlap chopper speed to less than 150Hz, the physical limit; changed the slot dimensions of the final [resolution] chopper, since LET now has a new disk; changed the frequency of chopper 4 to reflect new high resolution operating conditions.

3. Added facility to output a text to the screen similar to what the old matlab programs (`MChop` and `MultiRep`) used to do - particularly, outputs separately the time-widths at the detector due to the moderator and choppers.

**To test:**

Check in Mantid that `Interfaces` -> `Direct` -> `PyChop` runs. 

Select `LET` as the instrument, and check that the frequencies list of the pulse removal chopper go up to 150Hz only, in 10Hz increments. Check that the resolution chopper frequencies still go to 300Hz (also in 10Hz increments).

Set the resolution chopper to run at 300Hz and the pulse removal chopper at 100Hz, set the incident energy to 1meV and select `High Resolution` mode and click calculate. Then select `Intermediate` mode and click calculate again. Check that the second calculation gives a noticeably worst resolution than the first (previous, LET had a miscut disk, such that the resolution in `High Resolution` mode is about the same as in `Intermediate` mode. 

Click `Show multireps` and then calculate. Check that reps above 30meV are not shown in the plots (to see what they would be, click `Options` -> `Instrument Scientist Mode` and then click on the now revealed `Script Output` tab (you will have to click `Calculate` again.

Set `Merlin` as the instrument, select the `G` chopper and set the frequency to 600Hz, and incident energy to 50meV. Click `Calculate` and check that reps below 7meV are not shown in the plots.

Check that otherwise, PyChop behaves as expected.



Fixes #18485.



---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
